sheets/api.py

```diff
@@ -330,42 +338,71 @@ def __init__(self):
         )
         self.coupon_request_sheet = spreadsheet.sheet1
 
-    def parsed_row_iterator(self):
+    @staticmethod
+    def create_coupons_from_rows(enumerated_data_rows):
         """
-        Generator for successfully-parsed rows in the coupon request sheet. Rows that fail parsing
-        are logged and skipped.
+        Parses the given data rows, creates coupons for the rows that have not been processed,
+        and returns information about which rows were successfully processed and which ones failed.
 
-        Yields:
-            tuple of (int, CouponRequestRow): The row number in the sheet paired with the parsed coupon request row
+        Args:
+            enumerated_data_rows (iterable of (int, list of str)): Row indices paired with a list of strings
+                representing the data in each row
+
+        Returns:
+            (list of ProcessedRequest, list of FailedRequest, list of ProcessedRequest):
+                A list of objects representing each request row that was processed,
+                a list of objects representing each request row that couldn't be processed,
+                and a list of objects representing each request row that is complete but does't say so in
+                the spreadsheet.
         """
-        enumerated_data_rows = enumerate(
-            get_data_rows(self.coupon_request_sheet),
-            start=coupon_request_sheet_spec.first_data_row,
-        )
+        processed_requests = []
+        failed_requests = []
+        unrecorded_complete_requests = []
         for row_index, row_data in enumerated_data_rows:
             try:
-                yield row_index, CouponRequestRow.parse_raw_data(row_data)
+                coupon_req_row = CouponRequestRow.parse_raw_data(row_index, row_data)
             except Exception as exc:
                 log.error(
                     "Coupon request row could not be parsed (row %d). Exception: %s"
-                    % row_index,
-                    str(exc),
+                    % (row_index, str(exc))
+                )
+                failed_requests.append(
```

Comment:
Can you add a test for this? https://codecov.io/gh/mitodl/mitxpro/pull/1349/diff#D7-370